scout/server/blueprints/variant/controllers.py

```diff
@@ -644,36 +644,32 @@ def build_clinvar_submission(store, request, institute_id, case_name, variant_id
     updated_submission = store.add_to_submission(open_submission["_id"], submission_objects)
 
 
-def clinvar_export(store, institute_id, case_name, variant_id):
+def clinvar_export(store, institute_obj, case_obj, variants):
     """Gather the required data for creating the clinvar submission form
 
     Args:
         store(scout.adapter.MongoAdapter)
-        institute_id(str): Institute ID
-        case_name(str): case ID
-        variant_id(str): variant._id
+        institute_obj(str): Institute ID
+        case_obj(str): case ID
+        variants(list): list of variant._id strings
 
     Returns:
         data(dict): all the required data (case and variant level) to pre-fill in fields
                     in the clinvar submission form
 
     """
-    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
     # If case diagnoses are a list of integers, convert into a list of dictionaries
     case_diagnoses = case_obj.get("diagnosis_phenotypes", [])
     if case_diagnoses and isinstance(case_diagnoses[0], int):
         case_obj = store.convert_diagnoses_format(case_obj)
-    pinned = [
-        store.variant(variant_id) or variant_id for variant_id in case_obj.get("suspects", [])
-    ]
-    variant_obj = store.variant(variant_id)
+    variant_objs = [store.variant(variant_id) or variant_id for variant_id in variants]
 
     # gather missing transcript info from entrez (refseq id version)
-    for pinned_var in pinned:
+    for var_obj in variant_objs:
         # Exclude variants that aren't loaded
-        if isinstance(pinned_var, str):
+        if isinstance(var_obj, str):
```

Comment:
Is `var_obj` a string (as tested and expected) or something else?

Reply:
I'll add a comment. basically when a case has a variant is an object, but if a case gets re-uploaded without a variant, this becomes a string

Reply:
Ah there is already a comment